envi/__init__.py

```diff
@@ -600,16 +601,28 @@ def setEmuSnap(self, snap):
         self.setRegisterSnap(regs)
         self.setMemorySnap(mem)
 
+    @contextlib.contextmanager
+    def snap(self):
+        '''
+        Utility function to try something with an emulator, and then revert it.
+        '''
+        snap = self.getEmuSnap()
+        yield snap
+        self.setEmuSnap(snap)
```

Comment:
may want to do `try...finally` here or document what happens in the case of exceptions

Reply:
:+1: